src/extrapolation/filled.jl

```diff
@@ -54,12 +54,13 @@ end
     return ret
 end
 
-@inline function Interpolations.gradient(etp::FilledExtrapolation{T}, x::Vararg{Number,N}) where {T,N}
+@inline function Interpolations.gradient(etp::FilledExtrapolation{T, N, ITP}, x::Vararg{Number, N}) where {T,N,ITP}
```

Comment:
Why do you need the `ITP` parameter here?
```suggestion
@inline function Interpolations.gradient(etp::FilledExtrapolation{T, N}, x::Vararg{Number, N}) where {T,N}
```

Reply:
I guess it was meant to use `ITP` instead of `typeof(itp)`

Reply:
`ITP == typeof(itp)`. I'm guessing that affects the call `Base._return_type` which in turn affects the return type of the method. Therefore, one might need to parameterize on `ITP` for type stability of the overall method.